aa_stripe/models.py

```diff
@@ -462,17 +469,17 @@ class StripeWebhook(models.Model):
 
     def _parse_coupon_notification(self, action):
         coupon_id = self.raw_data["data"]["object"]["id"]
+        created = timestamp_to_timezone_aware_date(self.raw_data["data"]["object"]["created"])
         if action == "created":
-            try:
-                StripeCoupon(coupon_id=coupon_id).save(force_retrieve=True)
-            except stripe.error.InvalidRequestError:
-                # do not fail in case the coupon has already been removed from Stripe before we received the webhook
-                pass
+            if StripeCoupon.objects.filter(coupon_id=coupon_id, created=created).exists():
```

Comment:
.get would be better here - exists calls count which is pretty expensive. And what if coupon is deleted?